src/components/group-common.js

```diff
@@ -104,76 +105,115 @@ export const NewUserModal = ajaxCaller(class NewUserModal extends Component {
+        onDismiss: () => this.setState({ confirmAddUser: false })
       }, [
-        label({ style: { margin: '0 2rem 0 0.25rem' } }, [`Can manage users (${adminLabel})`])
-      ]),
-      footer && div({ style: { marginTop: '1rem' } }, [footer]),
-      submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.danger() } }, [submitError]),
-      submitting && spinnerOverlay
-    ])
+        'Add ', b(userEmail), ' to the group anyway?',
+        busy && spinnerOverlay
+      ])],
+      () => h(Modal, {
+        onDismiss,
+        title,
+        okButton: h(ButtonPrimary, {
+          tooltip: Utils.summarizeErrors(errors),
+          onClick: () => this.addUser(),
+          disabled: errors
+        }, ['Add User'])
+      }, [
+        h(RequiredFormLabel, ['User email']),
+        h(AutocompleteSearch, {
+          autoFocus: true,
+          value: userEmail,
+          onChange: v => this.setState({ userEmail: v }),
+          renderSuggestion: suggestion => div({ style: styles.suggestionContainer }, [
+            div({ style: { flex: 1 } }, [
+              !canAdd(suggestion) && h(TooltipTrigger, {
+                content: 'Not a valid email address'
+              }, [
+                icon('warning-standard', { style: { color: colors.danger(), marginRight: '0.5rem' } })
+              ]),
+              suggestion
+            ])
+          ]),
+          onSuggestionSelected: selection => {
+            this.setState({ userEmail: selection })
+          },
+          onKeyDown: e => {
+            // 27 = Escape
+            if (e.which === 27 && !!userEmail) {
+              this.setState({ userEmail: '' })
+              e.stopPropagation()
+            }
+          },
+          suggestions,
+          style: { fontSize: 16 },
+          theme: { suggestion: { padding: 0 } }
+        }),
+        h(FormLabel, ['Role']),
+        h(LabeledCheckbox, {
+          checked: isAdmin,
+          onChange: () => this.setState({ roles: [isAdmin ? userLabel : adminLabel] })
+        }, [
+          label({ style: { margin: '0 2rem 0 0.25rem' } }, [`Can manage users (${adminLabel})`])
+        ]),
+        footer && div({ style: { marginTop: '1rem' } }, [footer]),
+        submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.danger() } }, [submitError]),
+        busy && spinnerOverlay
+      ])
+    )
+  }
+
+  addUser() {
+    const { addUnregisteredUser = false, ajax: { User }, onDismiss } = this.props
+    const { userEmail } = this.state
+
+    Utils.cond(
+      [addUnregisteredUser, async () => {
+        try {
+          this.setState({ busy: true })
+          await User.isUserRegistered(userEmail) ? this.submit() : this.setState({ busy: false, confirmAddUser: true })
+        } catch (error) {
+          this.setState({ busy: false })
+          reportError('Error adding user', error)
+          onDismiss()
+        }
+      }],
+      () => this.submit()
+    )
```

Comment:
Can we simplify the logic here? Something like:
```
if (addUnregisteredUser && !(await User.isUserRegistered(userEmail))) {
  this.setState({ confirmAddUser: true })
} else {
  this.submit()
}
```
(wrapped in error handling and busy state handling as needed)